pydra/engine/specs.py

```diff
@@ -35,6 +46,21 @@ class SpecInfo:
 class BaseSpec:
     """The base dataclass specs for all inputs and outputs."""
 
+    def __setattr__(self, name, value):
+        """changing settatr, so the converter and validator is run
+        if input is set after __init__
+        """
+        if inspect.stack()[1][3] == "__init__":  # or name.startswith("_"):
+            super().__setattr__(name, value)
+        else:
+            tp = attr.fields_dict(self.__class__)[name].type
+            # if the type has a converter, e.g., MultiInputObj
+            if hasattr(tp, "converter"):
+                value = tp.converter(value)
+            super().__setattr__(name, value)
+            # validate all fields that have set a validator
+            attr.validate(self)
```

Comment:
have you checked if this statement does nothing when validation is disabled globally?

Reply:
forgot, have to add an if statement

Reply:
it does not run if validation is disabled